ch_tools/chadmin/internal/object_storage/s3_iterator.py

```diff
@@ -23,8 +26,11 @@ def s3_object_storage_iterator(
     )
     bucket = s3.Bucket(disk.bucket_name)
 
-    for obj in bucket.objects.filter(Prefix=disk.prefix + object_name_prefix):
-        name: str = obj.key[len(disk.prefix) :]
+    if not path_prefix:
+        path_prefix = disk.prefix
+
+    for obj in bucket.objects.filter(Prefix=path_prefix + object_name_prefix):
+        name: str = obj.key[len(path_prefix) :]
 
         if _is_ignored(name):
```

Comment:
`clean` command ignores objects with `IGNORED_OBJECT_NAME_PREFIXES`. But when removing data from S3 for deleted shards, I don't think we need any exceptions. We rather want to delete everything under the specified prefix.

Reply:
Done with `skip_ignoring` parameter